### DISCUSSION TITLE
[QUESTION] How to create static PersistentVolume name instead pvc-{uuid}?

### DISCUSSION THREAD

**Discussion by alex-kovalchuk** [longhorn/longhorn#5841] [Q&A]
Hello. Can I add some parameters to the StorageClass or PVC to force Longhorn PersistentVolume and Volume to be created with a static name instead of pvc-{uuid}?

First, it will help me in GitOps. Secondly, in case of problems, I can easily find the desired volume.. for example, if I lose the K8s cluster and I only have data in the partitions. Thirdly, it just saves me from manually creating volumes in Longhorn.

Thanks!

**Comment by weizhe0422** [comment]
I think the name could not be static if using dynamic provisioning, since the PV object was created by Kubernetes when CreateVolume call performs in CSI plugin.

**Comment by migs35323** [comment]
I do also have this problem, can we get a reply if this can be improved in longhorn or do we require to request this feature to other parties (kubernetes, csi spec, etc)

and can/t this be done with annotations or labels?
the only "easy" way to identify a volume is longhorn telling to which workload is attached to...

**Reply by derekbit** [reply]
@migs35323 
The PV name is determined by the external-provisioner ([codes](https://github.com/kubernetes-csi/external-provisioner/blob/ef83836bc21e5a737ded6d0f37f8a5f6a97d8973/pkg/controller/controller.go#L626)). I think you can request the feature in the Kubernetes community.

**Comment by specious-logic** [comment]
I do not really think this is a valid reason. Both local-path and nfs-path provisioner trivially solve this problem by simply naming the volume(path) from the PVC name with
pathPattern: ${.PVC.namespace}-${.PVC.name}

The current Longhorn behavior makes it exceeding difficult if an app must be effectively redeployed as a new deployment but you want to attach it to an existing volume.

If doing this via GitOps there is effectively no idempotent way to do this unless you bypass automatic provisioning and separate the PVC, PV, and Longhorn Volume config from the app config.
